### PR TITLE
SEIRD model implemented

### LINKED ISSUE #20: SEIRD
@Toby-exe, you can use the SIRD model as a base to work off of; there's no need to implement everything from scratch. Copying and pasting the contents of SIRD should work, as long as you then make the necessary changes to symbol names and whatnot with the equations.

I think we might even simply take your plotting function and generalize it to work for all models, unless there is something model-specific about it (other than the title, which we can pass as an argument to `ggplot`).

### PR DESCRIPTION
**Pull request by Toby-exe**
Closes #20 

Refactored to work with new ui naming scheme and design aswell as changed all camelCase to sna
eke_case + many stylistic and design changes (Very similar to SIRD_solve.R)

I wanted to change file name to SEIRD_solve.R but I don't want to lose git history on the file for this pr, will either make a new pr for the name change or just push the name change after pr is merged.

### CODE REVIEW

File: R/seird_plot.R [hunk 1]
```diff
@@ -1,131 +1,117 @@
...
-          scale_x_continuous(expand = c(0, 0)) +
-          scale_y_continuous(expand = c(0, 0)) +
-          scale_color_identity(
-            breaks = "Blue",
-            labels = "Susceptible"
-          )
-      })
-    })
+plotPhasePlaneSEIRD <- function() {
+  ggplot2::ggplot(solveSEIRD(), aes(x = S)) +
```
Comment (bryce-carson) [inline R/seird_plot.R:86]: `aes()` requires `ggplot2::` namespace.

File: R/seird_plot.R [hunk 2]
```diff
@@ -1,131 +1,117 @@
...
-          geom_line(aes(x = time, y = R, color = "Green"), linewidth = 1.5) +
-          geom_line(aes(x = time, y = D, color = "Orange"), linewidth = 1.5) +
-          scale_color_identity(
-            name = "SEIRD", breaks = c("Blue", "Brown", "Red", "Green", "Orange"),
-            labels = c("Susceptible", "Exposed", "Infected", "Recovered", "Dead"), guide = "legend"
-          )
-      })
-    })
+plotSEIRD <- function() {
+  ggplot2::ggplot(solveSEIRD(), aes(x = time)) +
```
Comment (bryce-carson) [inline R/seird_plot.R:58]: Here also we need `ggplot2::aes(...)`

File: R/seird_plot.R [hunk 3]
```diff
@@ -1,131 +1,117 @@
...
+  S <- variables[1]
+  E <- variables[2]
+  I <- variables[3]
+  R <- variables[4]
+  D <- variables[5]
+  N <- variables[6]
+  q <- variables[7]
+  dS <- (parameters["muB"] * N) - (parameters["muD"] * S) - (parameters["beta"] * ((S * I) / (N^q)))
+  dE <- (parameters["beta"] * ((S * I) / (N^q))) - (parameters["gamma"] * E) - (parameters["muD"] * E)
+  dI <- (parameters["gamma"] * E) - (parameters["delta"] * I) - (parameters["delta"] * I) - (parameters["muD"] * I)
```
Comment (bryce-carson) [inline R/seird_plot.R:11]: Change the first appearance of `delta` to `sigma`, then this line is okay.

File: R/seird_plot.R [hunk 4]
```diff
@@ -1,131 +1,117 @@
...
+  dR <- (parameters["delta"] * I) - (parameters["muD"] * R)
```
Comment (bryce-carson) [inline R/seird_plot.R:12]: Change `delta` to `sigma`.

**Review by bryce-carson — CHANGES_REQUESTED** [review]
There are only a couple copy-paste errors in this one, but otherwise it's ready to go!

**Comment by ashokkrish** [comment]
Just curious about something. If in the future I want to add a Vaccinated (V) compartment and want to run a SIRV model or a SVEIRD model is this going be a massive code change? Sorry to spring this on you out of nowhere.

**Comment by Toby-exe** [comment]
> There are only a couple copy-paste errors in this one, but otherwise it's ready to go!

Please see most recent commit, I also made the same changes for the namespace issue for SIRD, see #24

File: R/seird_plot.R [hunk 5]
```diff
@@ -1,131 +1,117 @@
...
+    ggplot2::theme(legend.position = "bottom") +
+    ggplot2::ylab("Number of People") +
+    ggplot2::xlab("Time") +
+    ggplot2::scale_x_continuous(expand = c(0, 0)) +
+    ggplot2::scale_y_continuous(expand = c(0, 0)) +
+    ggplot2::geom_line(aes(x = time, y = S, color = "Blue"), linewidth = 1.5) +
+    ggplot2::geom_line(aes(x = time, y = E, color = "Brown"), linewidth = 1.5) +
+    ggplot2::geom_line(aes(x = time, y = I, color = "Red"), linewidth = 1.5) +
+    ggplot2::geom_line(aes(x = time, y = R, color = "Green"), linewidth = 1.5) +
+    ggplot2::geom_line(aes(x = time, y = D, color = "Orange"), linewidth = 1.5) +
```
Comment (bryce-carson) [inline R/seird_plot.R:76]: The `aes` call on these lines still needs the `ggplot2::`.

Comment (bryce-carson) [inline R/seird_plot.R:76]: These need the `ggplot2::` as well.

File: R/seird_plot.R [hunk 6]
```diff
@@ -1,131 +1,117 @@
...
-          scale_y_continuous(expand = c(0, 0)) +
-          scale_color_identity(
-            breaks = "Blue",
-            labels = "Susceptible"
-          )
-      })
-    })
+plotPhasePlaneSEIRD <- function() {
+  ggplot2::ggplot(solveSEIRD(), ggplot2::aes(x = S)) +
+    ggplot2::geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
```
Comment (bryce-carson) [inline R/seird_plot.R:87]: Line 87 also.

**Review by bryce-carson — CHANGES_REQUESTED** [review]

**Comment by bryce-carson** [comment]
Great work, @Toby-exe!

**Pull request merged**